### PR TITLE
PI-1579 CAS3 Referral submitted processing

### PR DESCRIPTION
**Pull request by stevomcallister**

### CODE REVIEW

File: projects/cas3-and-delius/deploy/values.yaml [hunk 1]
```diff
@@ -12,6 +11,9 @@ generic-service:
   namespace_secrets:
     common:
       SPRING_DATASOURCE_URL: DB_URL
+    as3-and-delius-client-credentials:
```
Comment (marcus-bcl) [inline projects/cas3-and-delius/deploy/values.yaml:14]:
```suggestion
    cas3-and-delius-client-credentials:
```

File: projects/cas3-and-delius/deploy/values-dev.yml [hunk 2]
```diff
@@ -9,6 +7,9 @@ generic-service:
 
   env:
     SENTRY_ENVIRONMENT: dev
+    SPRING_SECURITY_OAUTH2_CLIENT_PROVIDER_HMPPS-AUTH_TOKEN-URI: https://sign-in-dev.hmpps.service.justice.gov.uk/auth/oauth/token
```
Comment (marcus-bcl) [inline projects/cas3-and-delius/deploy/values-dev.yml:10]: Do we need to add the approved-premises-api URL in here too?

File: projects/cas3-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/entity/Contact.kt [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    val type: ContactType,
+
+    @Lob
+    var notes: String,
+
+    @Column(name = "contact_date")
+    var date: LocalDate,
+
+    @Column(name = "contact_start_time")
+    var startTime: ZonedDateTime,
```
Comment (marcus-bcl) [inline projects/cas3-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/entity/Contact.kt:51]: Few vars in here, can they be vals?

File: projects/cas3-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/entity/PersonManager.kt [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    val staffEmployeeId: Long,
+
+    @Column(name = "probation_area_id")
+    val probationAreaId: Long,
+
+    @Column(name = "active_flag", columnDefinition = "NUMBER")
+    val active: Boolean = true,
+
+    @Column(name = "soft_deleted", columnDefinition = "NUMBER", nullable = false)
+    var softDeleted: Boolean = false
```
Comment (marcus-bcl) [inline projects/cas3-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/entity/PersonManager.kt:36]: Same here, can this be a val?

File: projects/cas3-and-delius/deploy/database/access.yml [hunk 5]
```diff
@@ -2,8 +2,11 @@ database:
...
     username_key: /cas3-and-delius/db-username
     password_key: /cas3-and-delius/db-password
+    tables:
+      - audited_interaction
+      - contact
 
   audit:
     username: Cas3AndDelius
-    forename: Probation Integration # TODO change this to something meaningful for your service
+    forename: cas3
```
Comment (marcus-bcl) [inline projects/cas3-and-delius/deploy/database/access.yml:11]: This is the user-facing name on the contact, will user's know it by CAS3?  Or should we call it something like "Transitional Accommodation (CAS3)"

**Review by marcus-bcl — APPROVED** [review]
Few very minor things, but looks good

File: projects/cas3-and-delius/deploy/values-dev.yml (hunk 2 above)
Comment (stevomcallister) [reply projects/cas3-and-delius/deploy/values-dev.yml:10]: I didn't think so since we only use the url from the message?

Comment (marcus-bcl) [reply projects/cas3-and-delius/deploy/values-dev.yml:10]: Ah good point sorry!

**Pull request merged**